File: core-lib/src/store/referral.rs
use crate::utill::types::referral::ReferralData;

pub fn get_referral_data() -> ReferralData {
    ReferralData {
        total_referrals: 12,
        active_referrals: 8,
        total_earnings: 120.00,
        payout: 120.00,
        referral_code: "ACME125FT".to_string(),
    }
}

File: ios/invoice/invoice/Models/Referral.swift
//
//  Referral.swift
//  invoice
//
//  Created by Ajibola Awotide on 2026-07-10.
//

struct Referral: Codable {
    public var totalReferrals: UInt64
    public var activeReferrals: UInt64
    public var totalEarnings: Double
    public var payout: Double
    public var referralCode: String
    
    static func fetchReferralData() async throws -> Referral {
        return try await APIClient.shared.request(
            path: "/api/v1/referral/details",
            method: "GET",
            requiresAuth: true
        )
    }
}
